checkov/common/util/parser_utils.py

```diff
@@ -375,23 +377,26 @@ def get_tf_definition_key(nested_module: str, module_name: str, module_index: An
 
 
 def get_tf_definition_key_from_module_dependency(
-    path: str, module_dependency: str | None, module_dependency_num: str | None
+    path: str, module_dependency: TFDefinitionKeyType | None, module_dependency_num: str | None
```

Comment:
this function should only get called from the old flow (that doesn't use TFDefinitionKeyType), why is this change needed?

Reply:
You're right. mypy error